src/worker.cpp

```diff
@@ -1,13 +1,266 @@
+#include "config.h"
+
 #include "worker.hpp"
 
+#include "exceptions.hpp"
+#include "logger.hpp"
+#include "utils.hpp"
+
+#include <filesystem>
+#include <fstream>
+#include <typeindex>
+
 namespace vpd
 {
-void Worker::selectDeviceTreeAndJson()
+
+Worker::Worker()
+{
+    auto jsonToParse = INVENTORY_JSON_DEFAULT;
+
+    // Check if symlink is already there to confirm fresh boot/factory
+    // reset.
+    if (std::filesystem::exists(INVENTORY_JSON_SYM_LINK))
+    {
+        jsonToParse = INVENTORY_JSON_SYM_LINK;
+        m_isSymlinkPresent = true;
+    }
+    // implies it is a fresh boot/factory reset.
+    else
+    {
+        // Create the directory for hosting the symlink
+        std::filesystem::create_directories(VPD_SYMLIMK_PATH);
+    }
+
+    std::ifstream inventoryJson(jsonToParse);
+    if (!inventoryJson)
+    {
+        throw(JsonException("Failed to access Json path", jsonToParse));
+    }
+
+    try
+    {
+        m_parsedJson = nlohmann::json::parse(inventoryJson);
+    }
+    catch (const nlohmann::json::parse_error& ex)
+    {
+        throw(JsonException("Json parsing failed", jsonToParse));
+    }
+}
+
+static std::string readFitConfigValue()
+{
+    std::vector<std::string> output = utils::executeCmd("/sbin/fw_printenv");
+    std::string fitConfigValue;
+
+    for (const auto& entry : output)
+    {
+        auto pos = entry.find("=");
+        auto key = entry.substr(0, pos);
+        if (key != "fitconfig")
+        {
+            continue;
+        }
+
+        if (pos + 1 < entry.size())
+        {
+            fitConfigValue = entry.substr(pos + 1);
+        }
+    }
+
+    return fitConfigValue;
+}
+
+bool Worker::isMotherboardPathOnDBus() const
```

Comment:
Will this function be used anywhere else, apart from setDevicetreeandJson()? If only there then may be a check should be good, I think in that function.

Reply:
What kind of check?

Reply:
motherboard path check

Reply:
That is what we are doing in this function. You want me to put all these code in that function itself?

Reply:
ok